File: Q1/codes/dia.c
// function to find the coordinates of point A
void find_coordinates(float Bx, float By, float Cx, float Cy, float *Ax, float *Ay) {
    *Ax = 2 * Cx - Bx;
    *Ay = 2 * Cy - By;
}

// Sample usage
#include <stdio.h>

int main() {
    float Bx = 2, By = 6;
    float Cx = 3, Cy = -1;
    float Ax, Ay;
    
    find_coordinates(Bx, By, Cx, Cy, &Ax, &Ay);

    printf("The coordinates of point A are: (%.2f, %.2f)\n", Ax, Ay);

    return 0;
}

File: Q1/codes/dia.py
# Function to find the coordinates of point A
def find_point_A(center, point_B):
    # Extract center and point_B coordinates
    center_x, center_y = center
    B_x, B_y = point_B
    
    # Use the midpoint formula to find A
    A_x = 2 * center_x - B_x
    A_y = 2 * center_y - B_y
    
    return A_x, A_y

# Given values
center = (3, -1)
point_B = (2, 6)

# Find coordinates of point A
point_A = find_point_A(center, point_B)
print("Coordinates of point A:", point_A)
